testkit/javadsl/src/main/scala/com/lightbend/lagom/javadsl/testkit/PersistentEntityTestDriver.scala

```diff
@@ -233,32 +241,31 @@ class PersistentEntityTestDriver[C, E, S](system: ActorSystem, entity: Persisten
   }
 
   private def checkSerialization(obj: Any): Option[Issue] = {
-    val obj1 = obj.asInstanceOf[AnyRef]
-    // check that it is configured
-    Try(serialization.findSerializerFor(obj1)) match {
-      case Failure(e) => Some(NoSerializer(obj, e))
-      case Success(serializer) =>
-        // verify serialization-deserialization round trip
-        Try(serializer.toBinary(obj1)) match {
-          case Failure(e) => Some(NotSerializable(obj, e))
-          case Success(blob) =>
-            val manifest = serializer match {
-              case ser: SerializerWithStringManifest => ser.manifest(obj1)
-              case _                                 => if (serializer.includeManifest) obj.getClass.getName else ""
-            }
-            serialization.deserialize(blob, serializer.identifier, manifest) match {
-              case Failure(e) => Some(NotDeserializable(obj, e))
-              case Success(obj2) =>
-                if (obj != obj2) {
-                  Some(NotEqualAfterSerialization(
-                    s"Object [$obj] does not equal [$obj2] after serialization/deserialization", obj1, obj2
-                  ))
-                } else if (serializer.isInstanceOf[JavaSerializer] && !isOkForJavaSerialization(obj1.getClass))
-                  Some(UsingJavaSerializer(obj1))
-                else
-                  None
-            }
-        }
+    Serialization.withTransportInformation(system.asInstanceOf[ExtendedActorSystem]) { () =>
+      val obj1 = obj.asInstanceOf[AnyRef]
+      // check that it is configured
+      Try(serialization.findSerializerFor(obj1)) match {
+        case Failure(e) => Some(NoSerializer(obj, e))
+        case Success(serializer) =>
+          // verify serialization-deserialization round trip
+          Try(serializer.toBinary(obj1)) match {
+            case Failure(e) => Some(NotSerializable(obj, e))
+            case Success(blob) =>
+              val manifest = Serializers.manifestFor(serializer, obj1)
```

Comment:
using the new utility `Serializers.manifestFor`, which does the same as the old code